lib/Auth/Source/Twitter.php

```diff
@@ -82,92 +71,98 @@ public function __construct(array $info, array $config)
         }
 
-        if (!isset($_REQUEST['oauth_verifier'])) {
+        $oauth_verifier = $request->get('oauth_verifier');
+        if ($oauth_verifier === null) {
             throw new Error\BadRequest("Missing oauth_verifier parameter.");
         }
-        $parameters['oauth_verifier'] = (string) $_REQUEST['oauth_verifier'];
 
-        $consumer = new Consumer($this->key, $this->secret);
-
-        Logger::debug("oauth: Using this request token [" .
-            $requestToken->key . "] with the secret [" . $requestToken->secret . "]");
+        $server = new TwitterServer(
+            [
+                'identifier' => $this->key,
+                'secret' => $this->secret,
+            ]
+        );
 
-        // Replace the request token with an access token
-        $accessToken = $consumer->getAccessToken(
-            'https://api.twitter.com/oauth/access_token',
+        $tokenCredentials = $server->getTokenCredentials(
             $requestToken,
-            $parameters
+            $request->get('oauth_token'),
+            $request->get('oauth_verifier')
         );
-        Logger::debug("Got an access token from the OAuth service provider [" .
-            $accessToken->key . "] with the secret [" . $accessToken->secret . "]");
 
-        $verify_credentials_url = 'https://api.twitter.com/1.1/account/verify_credentials.json';
-        if ($this->include_email) {
-            $verify_credentials_url = $verify_credentials_url . '?include_email=true';
-        }
-        $userdata = $consumer->getUserInfo($verify_credentials_url, $accessToken);
+        $state['token_credentials'] = serialize($tokenCredentials);
+        $userdata = $server->getUserDetails($tokenCredentials);
 
-        if (!isset($userdata['id_str']) || !isset($userdata['screen_name'])) {
-            throw new Error\AuthSource(
-                $this->authId,
-                'Authentication error: id_str and screen_name not set.'
-            );
+        $attributes = [];
+
+        foreach ($userdata->getIterator() as $key => $value) {
+            if (is_string($value)) {
+                $attributes['twitter.' . $key] = [$value];
+            }
         }
 
-        $attributes = [];
-        foreach ($userdata as $key => $value) {
+        foreach ($userdata->extra as $key => $value) {
             if (is_string($value)) {
                 $attributes['twitter.' . $key] = [$value];
             }
         }
 
-        $attributes['twitter_at_screen_name'] = ['@' . $userdata['screen_name']];
-        $attributes['twitter_screen_n_realm'] = [$userdata['screen_name'] . '@twitter.com'];
-        $attributes['twitter_targetedID'] = ['http://twitter.com!' . $userdata['id_str']];
+        $attributes['twitter_at_screen_name'] = ['@' . $userdata->nickname];
+        $attributes['twitter_screen_n_realm'] = [$userdata->nickname . '@twitter.com'];
+        $attributes['twitter_targetedID'] = ['http://twitter.com!' . $userdata->uid];
```

Comment:
I understand that this was already there. But are they useful/necessary? Their names are not in the `twitter.` namespace... but for BC you can keep as is of course.

Reply:
Removed them